blockstack/blockstackd.py

```diff
@@ -1023,6 +1023,28 @@ def rpc_get_blockstack_ops_hash_at( self, block_id, **con_info ):
         return self.success_response( {'ops_hash': ops_hash} )
 
 
+    def rpc_get_transaction_status(self, txid, **con_info):
+        """
+        Get the status of a transaction.
+        Return {'status': 'accepted', 'block_height': ..., 'vtxindex': ..., 'op': ...} if the transaction was accepted
+        Return {'status': 'rejected', 'block_height': ..., 'vtxindex': ..., 'op': ...} if the transaction was rejected
```

Comment:
What would be the status for the transactions:
- accepted / legit & unconfirmed?
- rejected / invalid & unconfirmed?
(by unconfirmed, I mean `tx.block_height + 6 > chain_tip.height` )

Reply:
Blockstack Core does not and cannot reasonably be modified to track unconfirmed transactions.

Reply:
Understand. Does that the endpoint would be returning  `{'status': 'ignored'}` as long as the transaction is not confirmed? If so, should we nuance the wording?

Reply:
Because Blockstack Core (1) does not track unconfirmed transactions, (2) does not pay attention to transactions that do not encode well-formed Blockstack transactions, and (3) does not track rejected transactions until explicitly told to do so, there is no way for it to distinguish between an unconfirmed transaction, a non-Blockstack transaction, and a rejected transaction that was not logged.